### PR TITLE
[release-4.6] Update to Go 1.15

### PR DESCRIPTION
**Pull request by LorbusChris**
Backports https://github.com/openshift/windows-machine-config-operator/pull/179 to release-4.6 branch.

### CODE REVIEW

**Comment by aravindhp** [comment]
I would rather not have a PR that is moving to Go 1.15 also adding a bug fix.

**Review by aravindhp — CHANGES_REQUESTED** [review]
Same comments here https://github.com/openshift/windows-machine-config-operator/pull/179#pullrequestreview-532420483 https://github.com/openshift/windows-machine-config-operator/pull/179#discussion_r525196810 here

**Comment by LorbusChris** [comment]
/hold
until #179 and #182 have merged. Will rebase after #182.

File: build/Dockerfile.ci
Comment (aravindhp) [inline build/Dockerfile.ci:8]: @LorbusChris please check with ART if they will be updating this image too. If they are not please open a ticket requesting them to do so.

**Comment by LorbusChris** [comment]
`registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6` now contains Go 1.15.2. I've updated the PR

**Review by aravindhp — APPROVED** [review]
/lgtm

**Pull request merged**
